Lightyear/Core.idr

```diff
@@ -129,34 +180,55 @@ commitTo (PT f) = PT $ \r, us, cs, ue, ce => f r cs cs ce ce
 interface Stream tok str | str where
   uncons : str -> Maybe (tok, str)
 
+  updatePos : Nat -> Position -> tok -> Pair Position Position
```

Comment:
I'm not sure if `updatePos` belongs in this interface when it does not mention `str` at all.

Shouldn't it go in an interface `Token tok` or something similar?